### PR TITLE
Python code: Fix more bare exceptions

### PR DESCRIPTION
**Pull request by bje-**
## What does this PR do?

Silences more E722 warnings ("do not use bare exceptions"). Many of these changes eliminate the exception catching around `GetDriverByName` as this function never raises an exception if the argument is a string, even if the driver does not exist. This simplifies the code a lot.

### CODE REVIEW

File: autotest/pymod/gdaltest.py [hunk 1]
```diff
@@ -1534,53 +1516,35 @@ def reregister_all_jpeg2000_drivers():
...
+        gdal.Debug('gdaltest', 'Registering JP2ECW')
+
+    if jp2mrsid_drv_unregistered:
+        jp2mrsid_drv.Register()
+        jp2mrsid_drv_unregistered = False
+        gdal.Debug('gdaltest', 'Registering JP2MrSID')
+
+    if jp2openjpeg_drv_unregistered:
+        jp2openjpeg_drv.Register()
+        jp2openjpeg_drv = False
```
Comment (rouault) [inline autotest/pymod/gdaltest.py]: should be jp2openjpeg_drv_unregistered = False

File: autotest/pymod/gdaltest.py [hunk 2]
```diff
@@ -1534,53 +1516,35 @@ def reregister_all_jpeg2000_drivers():
...
+        gdal.Debug('gdaltest', 'Registering JP2OpenJPEG')
+
+    if jp2lura_drv_unregistered:
+        jp2lura_drv.Register()
+        jp2lura_drv = False
```
Comment (rouault) [inline autotest/pymod/gdaltest.py]: should be jp2lura_drv_unregistered = False

Comment (bje-) [reply autotest/pymod/gdaltest.py]: Ditto here.

**Comment by rouault** [comment]
on mingw-w64:
```
  TEST: wms_5 ... fail (blowup)
Traceback (most recent call last):

  File "pymod\gdaltest_python2.py", line 43, in run_func

    result = func()

  File ".\wms.py", line 165, in wms_5

    if gdaltest.wms_drv is None:

AttributeError: 'module' object has no attribute 'wms_drv'
``

File: autotest/gdrivers/wms.py [hunk 3]
```diff
@@ -47,12 +47,7 @@
 
 def wms_1():
 
-    try:
-        gdaltest.wms_drv = gdal.GetDriverByName('WMS')
-    except:
-        gdaltest.wms_drv = None
-
-    if gdaltest.wms_drv is None:
+    if gdal.GetDriverByName('WMS') is None:
```
Comment (rouault) [inline autotest/gdrivers/wms.py]: Should be
```
gdaltest.wms_drv = gdal.GetDriverByName('WMS')
if gdaltest.wms_drv is None;
```

**Comment by rouault** [comment]
Ah sorry you copied my initial wrong version of the correction. It should be
```
gdaltest.wms_drv = gdal.GetDriverByName('WMS')
if gdaltest.wms_drv is None:
```
so with a column, not a semi-column

**Pull request merged**